runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,353 @@
+		}
+		defer func() {
+			err = c.dockerClient.ContainerRemove(ctx, response.ID, types.ContainerRemoveOptions{})
+			if err != nil {
+				_ = fmt.Errorf("failed to remove helper container. Provider=%s: %w", models.Docker, err)
+			}
+		}()
+		return c.dockerClient.ContainerExport(ctx, response.ID)
+	default:
+		return nil, fmt.Errorf("get raw contents not implemented for current object type (%s). Provider=%s", models.Docker, objectType)
+	}
+}
+
+func (c *Client) createScanConfigFile(config *provider.ScanJobConfig) error {
+	scanConfigFilePath, err := getScanConfigFileName(config)
+	if err != nil {
+		return err
+	}
+
+	_, err = os.Stat(scanConfigFilePath)
+	if errors.Is(err, os.ErrNotExist) {
+		err = os.WriteFile(scanConfigFilePath, []byte(config.ScannerCLIConfig), 0644)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) createScanContainer(ctx context.Context, config *provider.ScanJobConfig) (string, error) {
+	scanName, err := getScanName(config)
+	if err != nil {
+		return "", err
+	}
+
+	containerId, err := c.getContainerIdFromContainerName(ctx, scanName)
+	if containerId != "" {
+		return containerId, nil
+	}
+
+	pl, err := c.dockerClient.ImagePull(ctx, config.ScannerImage, types.ImagePullOptions{})
+	if err != nil {
+		return "", err
+	}
+	_, _ = io.Copy(io.Discard, pl)
+	_ = pl.Close()
+
+	scanConfigFilePath, err := getScanConfigFileName(config)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := c.dockerClient.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image: config.ScannerImage,
+			Entrypoint: []string{"sh", "-c",
+				fmt.Sprintf(
+					"/app/vmclarity-cli --config /tmp/%s --server %s --asset-scan-id %s",
+					filepath.Base(scanConfigFilePath),
+					config.VMClarityAddress,
+					config.AssetScanID,
+				),
+			},
+		},
+		&container.HostConfig{
+			Binds: []string{fmt.Sprintf("%s:/tmp", path.Dir(scanConfigFilePath))},
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeVolume,
+					Source: scanName,
+					Target: "/mnt/snapshot",
+				},
+			},
+		},
+		nil,
+		nil,
+		scanName,
+	)
+	if err != nil {
+		return "", err
+	}
+
+	return resp.ID, nil
+}
+
+func (c *Client) getContainerIdFromContainerName(ctx context.Context, scanName string) (string, error) {
+
+	containers, err := c.dockerClient.ContainerList(ctx, types.ContainerListOptions{
+		All:     true,
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to get containers. Provider=%s: %w", models.Docker, err)
+	}
+	if len(containers) == 0 {
+		return "", fmt.Errorf("scan container not found. Provider=%s: %w", models.Docker, err)
+	}
+	if len(containers) > 1 {
+		return "", fmt.Errorf("found more than one scan container. Provider=%s: %w", models.Docker, err)
+	}
+	return containers[0].ID, nil
+}
+
+func (c *Client) RemoveAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
```

Comment:
Can we have this public method before all private functions/methods. Usually, I like to follow the style-guide (top to bottom declarations):
1. Public constants and variables
2. Public functions
3. Public types
   - Related public methods first, then private methods
4. Private types
    - Related methods
 5. Private functions